tools/codegen/diff_native_functions.py

```diff
@@ -0,0 +1,143 @@
+import argparse
+import os
+import requests  # TODO(jwtan): Figure out the conventional API to use within the PyTorch repository.
+import subprocess
+import sys
+import tempfile
+
+# TODO(jwtan): Figure out why we need this.
+root = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+sys.path.insert(0, root)
+
+from tools.codegen.gen import parse_native_yaml
+
+# Returns the current_version and the base_version if specified by users.
+def get_versions() -> (str, str):
+    # TODO(jwtan): It returns a lot of information including release notes. Figure out a way to only query the tag names.
+    # TODO(jwtan): per_page=100 is not future proof.
+    response = requests.get("https://api.github.com/repos/pytorch/pytorch/releases?per_page=100")
+    releases = [release['tag_name'] for release in response.json()]
+
+    current_version = releases[0]
+
+    parser = argparse.ArgumentParser(description='Diff native functions. Please run this script in the open-source repository.')
```

Comment:
This is not very well organized. Ordinarily, I'd expect argument parsing to happen at the top level main function, and then the results of argument parsing to be passed to subsidiary functions. Furthermore, I would expect a CLI tool to avoid performing network requests before it has finished parsing arguments.